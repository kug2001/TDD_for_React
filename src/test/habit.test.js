import React from 'react';
import { screen, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import Habit from '../components/habit';

describe('habit 컴포넌트에 대한 테스트 코드', () => {
  const habit = { id: 1, name: 'sleep', count: 0 };
  let onIncrement;
  let onDecrement;
  let onDelete;
  beforeEach(() => {
    onIncrement = jest.fn();
    onDecrement = jest.fn();
    onDelete = jest.fn();
    render(<Habit habit={habit} onIncrement={onIncrement} onDecrement={onDecrement} onDelete={onDelete} />);
  })
  it('habit 컴포넌트 스냅샷 테스트', () => {
    const component = renderer.create(<Habit habit={habit} onIncrement={onIncrement} onDecrement={onDecrement} onDelete={onDelete} />);
    expect(component.toJSON()).toMatchSnapshot();
  })
  describe('habit 컴포넌트 테스트', () => {
    it('increase', () => {
      const handleIncrement = screen.getByTestId('increase');
      userEvent.click(handleIncrement);
      checkPropsFunctions(onIncrement, habit);
    })
    it('decrease', () => {
      const handleDecrement = screen.getByTestId('decrease');
      userEvent.click(handleDecrement);
      checkPropsFunctions(onDecrement, habit);
    })
    it('delete', () => {
      const handleDelete = screen.getByTestId('delete');
      userEvent.click(handleDelete);
      checkPropsFunctions(onDelete, habit);
    })
    function checkPropsFunctions(fn, habit) {
      expect(fn).toHaveBeenCalledTimes(1);
      expect(fn).toHaveBeenCalledWith(habit);
    }
  });
})

