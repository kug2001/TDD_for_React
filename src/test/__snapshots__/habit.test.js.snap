// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`habit 컴포넌트에 대한 테스트 코드 habit 컴포넌트 스냅샷 테스트 1`] = `
<li
  className="habit"
>
  <span
    className="habit-name"
  >
    sleep
  </span>
  <span
    className="habit-count"
    data-testid="count"
  >
    0
  </span>
  <button
    className="habit-button habit-increase"
    data-testid="increase"
    onClick={[Function]}
  >
    <i
      className="fas fa-plus-square"
    />
  </button>
  <button
    className="habit-button habit-decrease"
    data-testid="decrease"
    onClick={[Function]}
  >
    <i
      className="fas fa-minus-square"
    />
  </button>
  <button
    className="habit-button habit-delete"
    data-testid="delete"
    onClick={[Function]}
  >
    <i
      className="fas fa-trash"
    />
  </button>
</li>
`;
